public/js/endpoints.js

```diff
@@ -61,4 +61,116 @@ $(document).ready(function() {
             }
         });
 
-    });});
\ No newline at end of file
+    });});
+document.addEventListener('DOMContentLoaded', function () {
+    const checkboxes = document.querySelectorAll('.endpoint-checkbox');
+    const deleteSelectedBtn = document.getElementById('deleteSelected');
+    const installChipsecSelectedBtn = document.getElementById('installChipsecSelected');
+    const addNodeForm = document.getElementById('addNodeForm');
+
+    checkboxes.forEach(checkbox => {
+        checkbox.disabled = false;
+        checkbox.addEventListener('change', () => {
+            const selectedCheckboxes = document.querySelectorAll('.endpoint-checkbox:checked');
+            deleteSelectedBtn.disabled = selectedCheckboxes.length === 0;
+            installChipsecSelectedBtn.disabled = selectedCheckboxes.length === 0;
+        });
+    });
+
+    deleteSelectedBtn.addEventListener('click', () => {
+        const selectedCheckboxes = document.querySelectorAll('.endpoint-checkbox:checked');
+        const selectedIds = Array.from(selectedCheckboxes).map(cb => cb.dataset.uuid);
+        // handle delete action
+    });
+
+    installChipsecSelectedBtn.addEventListener('click', () => {
+        const selectedCheckboxes = document.querySelectorAll('.endpoint-checkbox:checked');
+        const selectedIds = Array.from(selectedCheckboxes).map(cb => cb.dataset.uuid);
+        // handle install chipsec action
+    });
+
+    addNodeForm.addEventListener('submit', (e) => {
+        e.preventDefault();
+        const osType = document.getElementById('osType').value;
+        const hostId = document.getElementById('hostId').value;
+
+        // Send AJAX request to add the new host
+        $.ajax({
+            url: '/endpoints.js/addNode',
+            method: 'POST',
+            data: {
+                osType,
+                hostId
+            },
+            success: function(response) {
+                // Handle success (e.g., refresh the list of endpoints)
+                location.reload();
+            },
+            error: function(error) {
+                // Handle error
+                console.error('Error adding host:', error);
+            }
+        });
+    });
+});
```

Comment:
Remove all these lines. Your already have an event listener for submit on line 139.
Also this ajax url is wrong (it has `.js` in the url)

Reply:
Got it, now i also understand Dar's resposne.
deleting and merging.